moby/moby
# Update DNS library

supersedes https://github.com/moby/moby/pull/40534
closes https://github.com/moby/moby/pull/40534

This makes sure that we don't become vulnerable to [CVE-2018-17419](https://nvd.nist.gov/vuln/detail/CVE-2018-17419) or
[CVE-2019-19794](https://nvd.nist.gov/vuln/detail/CVE-2019-19794) in the future. As far as I can tell we are *not* vulnerable to either right now, but if someone were to use one of the affected methods in the future we could become vulnerable with our current dependency.


**- What I did**

Bumped the version of `github.com/miekg/dns` to v1.1.27 in `vendor.conf`.

**- How I did it**

Manually edited the file, then ran the version of vendor installed in our containers by default to update the vendor tree.

eg. `go get -u github.com/LK4D4/vndr@85886e1ac99b8d96590e6e0d9f075dc7a711d132`

**- How to verify it**

CI should do that, I'm waiting on that now.


This also updates libnetwork to 264bffcb88c1b6b7471c04e3c6b3f301233a544b;

full diff: https://github.com/docker/libnetwork/compare/feeff4f0a3fd2a2bb19cf67c826082c66ffaaed9...264bffcb88c1b6b7471c04e3c6b3f301233a544b

includes:

- docker/libnetwork#2317 Allow bridge net driver to skip IPv4 configuration of bridge interface
    - adds support for a `com.docker.network.bridge.inhibit_ipv4` label/configuration
    - fixes moby/moby#37430 Prevent bridge network driver from setting IPv4 address on bridge interface
- docker/libnetwork#2454 Support for com.docker.network.host_ipv4 driver label
    - fixes moby/moby#30053 Unable to choose outbound (external) IP for containers
- docker/libnetwork#2491 Improving load balancer performance
    - fixes moby/moby#35082 [SWARM] Very poor performance for ingress network with lots of parallel requests
- docker/libnetwork#2419 Fix internal macvlan network to work in swarm
    - fixes docker/libnetwork#2418 Internal macvlan network doesn't work in swarm
    - relates to / addresses part of https://github.com/moby/moby/issues/39392

**- Description for the changelog**


```markdown
- Add support for a `com.docker.network.bridge.inhibit_ipv4` driver label to skip IPv4 configuration of bridge interface docker/libnetwork#2317
- Add support for a ` com.docker.network.host_ipv4` driver label to choose outbound (external) IP for containers docker/libnetwork#2454
- Improve load balancer performance for ingress network with lots of parallel requests docker/libnetwork#2491
```

**- A picture of a cute animal (not mandatory but encouraged)**
